### ISSUE TITLE
Apply Code Style and linter to entire PHP codebase

### ISSUE THREAD

**Issue by dmohns** [EnAccess/micropowermanager#107]
Currently the code style is only enforced in `app/` directory. We should run it over the entire PHP codebase.

**Comment by pechtelt1** [comment]
Hi @dmohns,

I'm new to opensource and start learning more about PHP. I'd like to work on this.

My plan is to update the Composer scripts so they cover not only the `app/` directory but also `packages/` and `tests`. Additionally, I will modify the GitHub Actions workflow to reflect these changes.

Does this approach sound correct before I create a PR? Thanks for your feedback!

**Comment by dmohns** [comment]
> I'm new to opensource and start learning more about PHP. I'd like to work on this.

Hey @pechtelt1 

happy to hear you are interested in Open Source and contributing to MPM. A warm welcome 😉 

> My plan is to update the Composer scripts so they cover not only the `app/` directory but also `packages/` and `tests`. Additionally, I will modify the GitHub Actions workflow to reflect these changes.
> 
> Does this approach sound correct before I create a PR? Thanks for your feedback!

Generally, the steps you listed are correct. However, I would like to break this down a bit further such that we can work on this task in small, incremental steps.

This is a rough list I have in mind

1. Add a new (temporary) composer script to lint PHP code outside of `app`.
2. Add a new (temporary) step in the Github Actions file to check for code style outside of `app`. When we first add this it will complain a lot, hence we have to put `continue-on-error: true`.
3. Apply the code style fixes
4. Remove  `continue-on-error: true` from the temporary Github Actions step
5. Merge the temporary composer and actions with the existing main ones.

Once all the above is done, there is a final but yet important step:

6. Add the commit hash of the refactor commit from 3. to `.git-blame-ignore-revs` to keep our blames clean.

Each of the above step should be a separate PR, so they are easy for us to review.

Please don't hesitate to reach out if you encounter any problems or have further questions. We recently launched an Open Source in Energy Access Discord server, feel free to join it if you prefer a more synchronous DM-style chat: https://community.oseas.org/

**Comment by pechtelt1** [comment]
Hi @dmohns, thanks for your reminder! I would really like to learn more about it and but haven't the time or aren't able to make the time for it to dive into it. Because have to finish some other projects. 

Sorry! Hope someone else wants to pick it up!

**Comment by dmohns** [comment]
No worries! I'm going to pick up the task myself then. You can always look at the corresponding Pull Requests to learn a little bit about it (They will be linked in this ticket).
